### PR TITLE
WIP on initial commit.

### PR DESCRIPTION
**Pull request by jeffmay**
@anxiousmodernman @egprentice @Nearhan @jtsmith0107

I'm working on writing a YAML library for Scala. Here is my work in progress.

The design of this API is to use an abstract syntax tree, similar to https://github.com/json4s/json4s-ast

Then do all operations over this tree in Scala, or serialize and deserialize using reader and writer typeclass instances.

I'm depending on [Snake YAML](https://bitbucket.org/asomov/snakeyaml/wiki/Documentation) (a fairly popular Java library) for parsing the YAML string into nodes, and then converting those nodes into `YamlValue`s instead of Java `Map`s or `List`s.

I'm also depending on [Scalactic](http://www.scalactic.org/) from the creators of ScalaTest for the ever readible and helpful `Or` monad and `Every` container to capture errors that accumulate from validation.
- [x] Needs cleaner error handling API
- [x] Needs to test all cases of YAML
- [x] Create test library for symmetric serialization property specs
- [ ] Needs README.md
- [x] Needs travis-ci
- [x] Needs coveralls
- [ ] Test coverage min > 80%

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -0,0 +1,50 @@
+organization in ThisBuild := "io.gloriousfuture"
+organizationName in ThisBuild := "The Glorious Future"
+scalaVersion in ThisBuild := "2.11.8"
```
Comment (egprentice) [inline build.sbt]: How about doing both 2.11.8 and 2.12.x ?

File: project/plugins.sbt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+addSbtPlugin("io.get-coursier" % "sbt-coursier" % "1.0.0-M14")
+addSbtPlugin("me.lessis" % "bintray-sbt" % "0.3.0")
```
Comment (egprentice) [inline project/plugins.sbt]: missing EOL at EOF

File: build.sbt (hunk 1 above)
Comment (jeffmay) [inline build.sbt]: When does 2.12 release?

File: project/plugins.sbt (hunk 2 above)
Comment (jeffmay) [inline project/plugins.sbt]: Fixed.

File: build.sbt (hunk 1 above)
Comment (egprentice) [inline build.sbt]: Soon. http://www.scala-lang.org/news/2.12.0-RC2

**Comment by jeffmay** [comment]
retest this please

**Comment by jeffmay** [comment]
Looks like much of what I need is actually provided by circle-yaml:
https://github.com/jeremyrsmith/circe-yaml

I'm going to merge this PR once tests are passing and what not, but I'm gonna shelve this project while I try out using circe.

**Comment by jeffmay** [comment]
I'm moving a lot of this work to circe-yaml, so that I don't need to create a separate `YamlValue` AST. Circe is built on cats and is a fork of the amazingly powerful Argonaut project.

If you are interested in reviewing the changes, they live here:
https://github.com/jeremyrsmith/circe-yaml/pull/3

**Pull request closed without merging**